classes/local/form/login_form.php

```diff
@@ -104,10 +69,8 @@ public function define_grace_period_page($mform) {
     public function validation($data, $files) {
         $errors = parent::validation($data, $files);
 
-        if (!$data['grace_mode']) {
-            $factor = \tool_mfa\plugininfo\factor::get_factor($data['factor_name']);
-            $errors += $factor->login_form_validation($data);
-        }
+        $factor = \tool_mfa\plugininfo\factor::get_factor($data['factor_name']);
```

Comment:
We don't seem to even need the $factor or the factor_name here? dead code from before?

Reply:
The factor is needed, to call the validation injection from the factor class